crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -14,18 +16,28 @@ use super::system_state_summary::SystemStateSummary;
 use super::transaction_block::{self, TransactionBlock, TransactionBlockFilter};
 use super::validator_set::ValidatorSet;
 use async_graphql::connection::Connection;
+use async_graphql::dataloader::{DataLoader, Loader};
 use async_graphql::*;
 use diesel::{ExpressionMethods, OptionalExtension, QueryDsl, SelectableHelper};
 use fastcrypto::encoding::{Base58, Encoding};
 use sui_indexer::models::epoch::QueryableEpochInfo;
 use sui_indexer::schema::epochs;
 use sui_types::messages_checkpoint::CheckpointCommitment as EpochCommitment;
 
+#[derive(Clone)]
 pub(crate) struct Epoch {
     pub stored: QueryableEpochInfo,
     pub checkpoint_viewed_at: Option<u64>,
 }
 
+/// DataLoader key for fetching an `Epoch` by its ID, optionally constrained by a consistency
+/// cursor.
+#[derive(Copy, Clone, Hash, Eq, PartialEq, Debug)]
```

Comment:
Why does this need to be Copy? Is it because of `Option`?

Reply:
`Copy` is used at the end of the data loader implementation, when creating the hashmap -- the keys are copied from the input.